Bug: The README examples don't compile

Compiling the example in the README results in the following build errors:

```shell
$ go build
# local/example
./main.go:17:10: undefined: ClientOptions
./main.go:18:12: undefined: NewClientWithOptions
./main.go:20:44: undefined: dr
./main.go:23:28: undefined: tx
```

--- a/README.md
+++ b/README.md
@@ -30,17 +30,17 @@ import (
 )
 
 func main() {
-    ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: "{{ACCESS_TOKEN}}"})
-    ctx := context.Background()
-    tc := oauth2.NewClient(ctx, ts)
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: "{{ACCESS_TOKEN}}"})
+	ctx := context.Background()
+	tc := oauth2.NewClient(ctx, ts)
 
-    client := starling.NewClient(tc)
+	client := NewClient(tc)
 
-    txns, _, _ := client.GetTransactions(ctx, dr)
+	txns, _, _ := client.Transactions(ctx, nil)
 
-    for i, txn := range txns.Transactions {
-        fmt.Println(txn.Created, tx.Amount, txn.Currency, txn.Narrative)
-    }
+	for _, txn := range txns {
+		fmt.Println(txn.Created, txn.Amount, txn.Currency, txn.Narrative)
+	}
 }
 ```
 
@@ -52,25 +52,26 @@ package main
 import (
     "context"
     "fmt"
+    "net/url"
 
     "github.com/billglover/starling"
     "golang.org/x/oauth2"
 )
 
 func main() {
-    ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: "{{ACCESS_TOKEN}}"})
-    ctx := context.Background()
-    tc := oauth2.NewClient(ctx, ts)
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: "{{ACCESS_TOKEN}}"})
+	ctx := context.Background()
+	tc := oauth2.NewClient(ctx, ts)
 
-    baseURL, _ := url.Parse("https://dummyurl:4000")
-    opts := ClientOptions{BaseURL: baseURL}
-    client := NewClientWithOptions(nil, opts)
+	baseURL, _ := url.Parse(ProdURL)
+	opts := ClientOptions{BaseURL: baseURL}
+	client := NewClientWithOptions(tc, opts)
 
-    txns, _, _ := client.GetTransactions(ctx, dr)
+	txns, _, _ := client.Transactions(ctx, nil)
 
-    for i, txn := range txns.Transactions {
-        fmt.Println(txn.Created, tx.Amount, txn.Currency, txn.Narrative)
-    }
+	for _, txn := range txns {
+		fmt.Println(txn.Created, txn.Amount, txn.Currency, txn.Narrative)
+	}
 }
 ```
 

--- a/README.md
+++ b/README.md
@@ -2,14 +2,18 @@
 
 [![Build Status](https://travis-ci.com/billglover/starling.svg?branch=master)](https://travis-ci.com/billglover/starling)
 
-This is a Go client for the Starling Bank API.
+This is an unofficial Go client for the Starling Bank API.
 
-Both the API itself and this client are under active development and cannot guarantee a stable interface. A versioning approach has not yet been determined and so breaking changes are possible. That said, the client is in active use by a number of small projects.
+Both the Starling Bank API itself and this package are under active development and, whilst we try to keep breaking changes to a minimum, we cannot guarantee a stable interface. We use [Semantic Versioning](https://semver.org) to quantify changes from one release to the next.
+
+> "Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable."
 
 ## Installation
 
+Use Go to fetch the latest version of the package.
+
 ```shell
-go get 'github.com/billglover/starling'
+go get -u 'github.com/billglover/starling'
 ```
 
 ## Usage
@@ -70,63 +74,6 @@ func main() {
 }
 ```
 
-## Starling Developer Documentation
+## Starling Bank Developer Documentation
 
 * [Developer Documentation](https://developer.starlingbank.com/)
-
-## Features
-
-### API Coverage
-
-| Method | Resource                                                              | Status      |
-|--------|-----------------------------------------------------------------------|------------:|
-| PUT    | /api/v1/savings-goals/{savingsGoalUid}/add-money/{transferUid}        | Done        |
-| GET    | /api/v1/savings-goals/{savingsGoalUid}                                | Done        |
-| PUT    | /api/v1/savings-goals/{savingsGoalUid}                                | Done        |
-| DELETE | /api/v1/savings-goals/{savingsGoalUid}                                | Done        |
-| GET    | /api/v1/savings-goals                                                 | Done        |
-| GET    | /api/v1/savings-goals/{savingsGoalUid}/photo                          | Done        |
-| PUT    | /api/v1/savings-goals/{savingsGoalUid}/withdraw-money/{transferUid}   | Done        |
-| GET    | /api/v1/savings-goals/{savingsGoalUid}/recurring-transfer             | Done        |
-| PUT    | /api/v1/savings-goals/{savingsGoalUid}/recurring-transfer             | Done        |
-| DELETE | /api/v1/savings-goals/{savingsGoalUid}/recurring-transfer             | Done        |
-| GET    | /api/v1/accounts                                                      | Done        |
-| GET    | /api/v1/accounts/balance                                              | Done        |
-| GET    | /api/v1/addresses                                                     | Done        |
-| GET    | /api/v1/contacts                                                      | Done        |
-| POST   | /api/v1/contacts                                                      | Done        |
-| GET    | /api/v1/contacts/{id}                                                 | Done        |
-| DELETE | /api/v1/contacts/{id}                                                 | Done        |
-| GET    | /api/v1/contacts/{id}/accounts                                        | Done        |
-| GET    | /api/v1/contacts/{contactId}/accounts/{accountId}                     | Done        |
-| GET    | /api/v1/customers                                                     | Done        |
-| GET    | /api/v1/direct-debit/mandates                                         | Done        |
-| GET    | /api/v1/direct-debit/mandates/{mandateUid}                            | Done        |
-| DELETE | /api/v1/direct-debit/mandates/{mandateUid}                            | Done        |
-| GET    | /api/v1/me                                                            | Done        |
-| GET    | /api/v1/cards                                                         | Done        |
-| GET    | /api/v1/merchants/{merchantUid}                                       | Done        |
-| GET    | /api/v1/merchants/{merchantUid}/locations/{merchantLocationUid}       | Done        |
-| POST   | /api/v1/payments/local                                                | Done        |
-| GET    | /api/v1/payments/scheduled                                            | Done        |
-| POST   | /api/v1/payments/scheduled                                            | Done        |
-| GET    | /api/v1/transactions/direct-debit                                     | Done        |
-| GET    | /api/v1/transactions/direct-debit/{transactionUid}                    | Done        |
-| PUT    | /api/v1/transactions/direct-debit/{transactionUid}                    | Done        |
-| GET    | /api/v1/transactions/fps/in                                           | Done        |
-| GET    | /api/v1/transactions/fps/in/{transactionUid}                          | Done        |
-| GET    | /api/v1/transactions/fps/out                                          | Done        |
-| GET    | /api/v1/transactions/fps/out/{transactionUid}                         | Done        |
-| GET    | /api/v1/transactions/mastercard                                       | Done        |
-| GET    | /api/v1/transactions/mastercard/{transactionUid}                      | Done        |
-| PUT    | /api/v1/transactions/mastercard/{transactionUid}                      | Done        |
-| POST   | /api/v1/transactions/mastercard/{transactionUid}/receipt              | Done        |
-| PUT    | /api/v1/transactions/mastercard/{transactionUid}/receipt/{receiptUid} | Invalid     |
-| GET    | /api/v1/transactions                                                  | Done        |
-| GET    | /api/v1/transactions/{transactionUid}                                 | Done        |
-
-### Webhook Coverage
-
-| Method | Resource                                                            | Status      |
-|--------|---------------------------------------------------------------------|------------:|
-| POST   | /your-registered-web-hook-address/card-transaction                  |             |

--- a/README.md
+++ b/README.md
@@ -18,6 +18,8 @@ go get -u 'github.com/billglover/starling'
 
 ## Usage
 
+It is assumed that you are able to provide an OAuth access-token when establishing the Starling client. Depending on your use case, it pay be as simple as passing in the personal access-token provided by Starling when you create an applicaiton. See the section on Personal Access Tokens in the [Starling Developer Docs](https://developer.starlingbank.com/docs) for more information on how to do this.
+
 ```go
 package main
 

--- a/example_test.go
+++ b/example_test.go
@@ -1,0 +1,43 @@
+package starling
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"golang.org/x/oauth2"
+)
+
+// Note: the examples listed here are compiled but not executed while testing.
+// See the documentation on [Testing](https://golang.org/pkg/testing/#hdr-Examples)
+// for further details.
+
+func Example_sandbox() {
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: "{{ACCESS_TOKEN}}"})
+	ctx := context.Background()
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := NewClient(tc)
+
+	txns, _, _ := client.Transactions(ctx, nil)
+
+	for _, txn := range txns {
+		fmt.Println(txn.Created, txn.Amount, txn.Currency, txn.Narrative)
+	}
+}
+
+func Example_live() {
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: "{{ACCESS_TOKEN}}"})
+	ctx := context.Background()
+	tc := oauth2.NewClient(ctx, ts)
+
+	baseURL, _ := url.Parse(ProdURL)
+	opts := ClientOptions{BaseURL: baseURL}
+	client := NewClientWithOptions(tc, opts)
+
+	txns, _, _ := client.Transactions(ctx, nil)
+
+	for _, txn := range txns {
+		fmt.Println(txn.Created, txn.Amount, txn.Currency, txn.Narrative)
+	}
+}

--- a/example_test.go
+++ b/example_test.go
@@ -1,10 +1,11 @@
-package starling
+package starling_test
 
 import (
 	"context"
 	"fmt"
 	"net/url"
 
+	"github.com/billglover/starling"
 	"golang.org/x/oauth2"
 )
 
@@ -17,7 +18,7 @@ func Example_sandbox() {
 	ctx := context.Background()
 	tc := oauth2.NewClient(ctx, ts)
 
-	client := NewClient(tc)
+	client := starling.NewClient(tc)
 
 	txns, _, _ := client.Transactions(ctx, nil)
 
@@ -31,9 +32,9 @@ func Example_live() {
 	ctx := context.Background()
 	tc := oauth2.NewClient(ctx, ts)
 
-	baseURL, _ := url.Parse(ProdURL)
-	opts := ClientOptions{BaseURL: baseURL}
-	client := NewClientWithOptions(tc, opts)
+	baseURL, _ := url.Parse(starling.ProdURL)
+	opts := starling.ClientOptions{BaseURL: baseURL}
+	client := starling.NewClientWithOptions(tc, opts)
 
 	txns, _, _ := client.Transactions(ctx, nil)
 

